### PR TITLE
Add files used in OpenShift CI.

### PR DESCRIPTION
**Pull request by tomob**
Signed-off-by: Tomasz Baranski <tbaransk@redhat.com>

**What this PR does / why we need it**:

Added file `cluster-sync/sync-os-ci.sh`, which pushes images to a registry and deploys CDI. Added missing commands in `hack/ci/Dockerfile.ci`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by maya-r** [comment]
Does the Makefile indentation really not mess things up?

File: cluster-sync/sync-os-ci.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# instead we use a local registry; so here we'll check for anything != "external"
+# wel also confuse this by swapping the setting of the DOCKER_PREFIX variable around based on it's context, for
+# build and push it's localhost, but for manifests, we sneak in a change to point a registry container on the
+# kubernetes cluster.  So, we introduced this MANIFEST_REGISTRY variable specifically to deal with that and not
+# have to refactor/rewrite any of the code that works currently.
+MANIFEST_REGISTRY=$DOCKER_PREFIX
+
+# Need to set the DOCKER_PREFIX appropriately in the call to `make docker push`, otherwise make will just pass in the default `kubevirt`
+DOCKER_PREFIX=$DOCKER_PREFIX make push
+exit 0
```
Comment (maya-r) [inline cluster-sync/sync-os-ci.sh:34]: exit 0 mid-script seems to make a lot of the lines below to be dead code

Comment (tomob) [reply cluster-sync/sync-os-ci.sh:34]: :man_facepalming: :smile:

**Comment by maya-r** [comment]
it's not immediately clear why so much of cluster-sync/sync.sh is duplicated, can you explain what the differences are in a comment?

**Comment by tomob** [comment]
@maya-r The sync script for OS CI needs to handle `external` Kubevirt provider differently. Also, I removed all code dealing with upgrades.

**Review by awels — COMMENTED** [review]
/approve

**Pull request merged**